### PR TITLE
Support diffing blueprints via Diffus

### PR DESCRIPTION
**Pull request by andrewjstone**
This is the second step in https://github.com/oxidecomputer/omicron/issues/7240. 
The next steps will use this support to replace our handrolled code for generating diffs and generate output from the diffus based diffs.

### CODE REVIEW

File: nexus/types/src/deployment.rs [hunk 1]
```diff
@@ -187,6 +190,7 @@ pub struct Blueprint {
     pub clickhouse_cluster_config: Option<ClickhouseClusterConfig>,
 
     /// when this blueprint was generated (for debugging)
+    #[diffus(ignore)]
```
Comment (jgallagher) [inline nexus/types/src/deployment.rs:193]: Do we also want to ignore `creator` and `comment`? I think we don't care about this in diffs either.

File: nexus/reconfigurator/planning/src/planner.rs [hunk 2]
```diff
@@ -909,6 +910,12 @@ mod test {
             "2 -> 3 (expect new NTP zone on new sled):\n{}",
             diff.display()
         );
+        let diffus_diff = blueprint3.diff(&blueprint2);
+        println!("_______DIFFUS __________\n");
+        let change = diffus_diff.change();
+        println!("{:#?}", diffus_diff);
+        println!("{:#?}", change);
+        println!("________DIFFUS __________\n");
```
Comment (jgallagher) [inline nexus/reconfigurator/planning/src/planner.rs:918]: Just checking - intended to leave this here?

**Review by jgallagher — APPROVED** [review]
Easy enough! This seems a lot nicer than maintaining hand-roll diffs, to put it mildly.

File: nexus/types/src/deployment.rs (hunk 1 above)
Comment (andrewjstone) [reply nexus/types/src/deployment.rs:193]: After discussion with Rain, I'm going to write a visitor. I think it's worth allowing the possibility to diff these via visitor implementations.

File: nexus/reconfigurator/planning/src/planner.rs (hunk 2 above)
Comment (andrewjstone) [reply nexus/reconfigurator/planning/src/planner.rs]: I just removed it. Thanks!

**Comment by sunshowers** [comment]
This is wonderful -- thanks again for all the work and the demo @andrewjstone!

**Pull request merged**